src/kubeobjects/istio.go

```diff
@@ -0,0 +1,237 @@
+}
+
+func buildObjectMeta(name string, namespace string) metav1.ObjectMeta {
+	return metav1.ObjectMeta{
+		Name:      name,
+		Namespace: namespace,
+	}
+}
+
+func BuildIstioLabels(name, role string) map[string]string {
+	return map[string]string{
+		"dynatrace":            "oneagent",
+		"oneagent":             name,
+		"dynatrace-istio-role": role,
+	}
+}
+
+func isIp(host string) bool {
+	return net.ParseIP(host) != nil
+}
+
+func VerifyIstioCrdAvailability(instance *dynatracev1beta1.DynaKube, config *rest.Config) probeResult {
+	var probe probeResult
+
+	probe, _ = kubernetesObjectProbe(ServiceEntryGVK, instance.GetNamespace(), "", config)
+	if probe == ProbeTypeNotFound {
+		return probe
+	}
+
+	probe, _ = kubernetesObjectProbe(VirtualServiceGVK, instance.GetNamespace(), "", config)
+	if probe == ProbeTypeNotFound {
+		return probe
+	}
+
+	return ProbeTypeFound
+}
+
+func kubernetesObjectProbe(gvk schema.GroupVersionKind,
+	namespace string, name string, config *rest.Config) (probeResult, error) {
+
+	var objQuery unstructured.Unstructured
+	objQuery.Object = make(map[string]interface{})
+
+	objQuery.SetGroupVersionKind(gvk)
+
+	runtimeClient, err := client.New(config, client.Options{})
+	if err != nil {
+		return probeUnknown, err
+	}
+	if name == "" {
+		err = runtimeClient.List(context.TODO(), &objQuery, client.InNamespace(namespace))
+	} else {
+		err = runtimeClient.Get(context.TODO(), client.ObjectKey{Namespace: namespace, Name: name}, &objQuery)
+	}
+
+	return mapErrorToObjectProbeResult(err)
+}
+
+func mapErrorToObjectProbeResult(err error) (probeResult, error) {
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return probeObjectNotFound, err
+		} else if meta.IsNoMatchError(err) {
+			return ProbeTypeNotFound, err
+		}
+
+		return probeUnknown, err
+	}
+
+	return probeObjectFound, nil
+}
+
+func HandleIstioConfigurationForServiceEntry(instance *dynatracev1beta1.DynaKube,
+	name string, communicationHost dtclient.CommunicationHost, role string, config *rest.Config, namespace string,
+	istioClient istioclientset.Interface, scheme *runtime.Scheme, log logr.Logger) (bool, error) {
+
+	probe, err := kubernetesObjectProbe(ServiceEntryGVK, instance.GetNamespace(), name, config)
+	if probe == probeObjectFound {
+		return false, nil
+	} else if probe == probeUnknown {
+		//log.Error(err, "istio: failed to query ServiceEntry")
+		return false, err
+	}
+
+	serviceEntry := buildServiceEntry(name, namespace, communicationHost.Host, communicationHost.Protocol, communicationHost.Port)
+	err = createIstioConfigurationForServiceEntry(instance, serviceEntry, role, istioClient, scheme)
+	if err != nil {
+		//log.Error(err, "istio: failed to create ServiceEntry")
+		return false, err
+	}
+	log.Info("istio: ServiceEntry created", "objectName", name, "host", communicationHost.Host, "port", communicationHost.Port)
+
+	return true, nil
+}
+
+func HandleIstioConfigurationForVirtualService(instance *dynatracev1beta1.DynaKube,
```

Comment:
`virtualservice.go` under the `istio` package
you shouldn't pass in a logger, use the package level logger instead